### PR TITLE
fix: Resolve CocoaPods Version Issue and Ruby Installation Failure for Podfile.lock

### PR DESCRIPTION
**Pull request by Andepande**
## **Related issues**

Fixes: #
- Install ruby on bitrise env using asdf
- install sudo gem install cocoapods (may refactor later)
## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -952,7 +942,6 @@ SPEC CHECKSUMS:
   Yoga: e71803b4c1fff832ccf9b92541e00f9b873119b9
   YogaKit: f782866e155069a2cca2517aafea43200b01fd5a
 
+PODFILE CHECKSUM: 52f09b35897356bf28e08a4a40cae42c4f13fb23
 
-PODFILE CHECKSUM: 6c9103d9058ffd16683843f0f7d99bfa6f599ee3
-
-COCOAPODS: 1.12.1
+COCOAPODS: 1.11.3
```
Comment (OGPoyraz) [inline ios/Podfile.lock:947]: Does that mean we downgrade to `1.11.3`? If so can we update readme file pointing out to `1.12.1`?

**Comment by tommasini** [comment]
Can we update the PR description to say why we need the update? 
I'm not understanding the root cause of this issue, it suddenly start failing or am I missing something?

**Review by tommasini — APPROVED** [review]
LGTM! talking with @Andepande this issue seems to be due some bitrise lately update

**Review by OGPoyraz — APPROVED** [review]

File: ios/Podfile.lock (hunk 1 above)
Comment (tommasini) [reply ios/Podfile.lock]: Interesting, on our gemfile we do have the 1.11.3 as the minimum version https://github.com/MetaMask/metamask-mobile/blob/867a410643e69cb3e81acb8f667436767e88daa5/Gemfile#L6

I don't know why on readme we have 1.12.1

**Pull request merged**